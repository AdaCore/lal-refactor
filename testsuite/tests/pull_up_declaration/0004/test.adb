package body Test is
   ---------
   -- Foo --
   ---------

   procedure Foo is
      procedure Bar;
      --  Bar

      ---------
      -- Bar --
      ---------

      procedure Bar is
      begin
         null;
      end Bar;

   begin
      null;
   end Foo;
end Test;
