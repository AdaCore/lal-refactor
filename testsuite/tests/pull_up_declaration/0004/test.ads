package Test is
   procedure Foo;
   --  Foo
end Test;
