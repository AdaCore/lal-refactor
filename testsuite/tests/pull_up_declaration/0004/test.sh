pull_up_declaration -P test.gpr --source test.adb --line 7 --column 18
