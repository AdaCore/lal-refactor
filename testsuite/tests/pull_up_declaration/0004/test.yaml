title: pull_up_from_subp_body_with_header

description: |
    Pull Up Declaration tool test
    This test checks that when pulling up a declaration for a subprogram body
    with a header, the insertion point of the pulled up declaration is above
    the subprogram body header.

driver: shell_script
