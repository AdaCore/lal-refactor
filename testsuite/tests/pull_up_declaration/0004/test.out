test.adb
2:1-2:1    procedure Bar;
   --  Bar

   ---------
   -- Bar --
   ---------

   procedure Bar is
   begin
      null;
   end Bar;


7:1-18:1 

